mesheryctl/internal/cli/root/workspaces/create.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workspaces
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var CreateWorkspaceCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a new workspace",
+	Long:  `Create a new workspace by providing the name, description, and organization ID.`,
+	Example: `
+// Create a new workspace
+mesheryctl exp workspace create --orgId [orgId] --name [name] --description [description]
+
+// Documentation for workspace can be found at:
+https://docs.layer5.io/cloud/spaces/workspaces/
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		// Check prerequisites
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		if err := utils.IsServerRunning(mctlCfg.GetBaseMesheryURL()); err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return system.ErrGetCurrentContext(err)
+		}
+		if err := ctx.ValidateVersion(); err != nil {
+			return err
+		}
+		return nil
+	},
+
+	Args: func(cmd *cobra.Command, args []string) error {
+		// Check if all three flags are set
+		orgIdFlag, _ := cmd.Flags().GetString("orgId")
+		nameFlag, _ := cmd.Flags().GetString("name")
+		descriptionFlag, _ := cmd.Flags().GetString("description")
+
+		if orgIdFlag == "" || nameFlag == "" || descriptionFlag == "" {
+			if err := cmd.Usage(); err != nil {
+				return err
+			}
+			return errors.New("all three flags --orgId, --name, and --description are required")
```

Comment:
This error is not acceptable.
Define a ErrInvalidArgs (a generic err function that accepts invalid args)